Map/RoomBase.gd

```diff
@@ -1,21 +1,17 @@
 extends Resource
-class_name Room
+class_name RoomBase
 ## Class for a room on the map. Provides base classes for all rooms to use regardless of the type
 ##
 ## Holds basic functionality for a room on the map
 
-enum RoomType
+enum RoomEvent
 {
-	UNKNOWN,
 	MONSTER,
 	HEAL,
 	SHOP,
 }
 
 @export var light_level: int = 0
 @export var has_torch: bool = false
-@export var this_room_type:RoomType = RoomType.UNKNOWN
-@export var coordinates:Vector2
-func get_room_type():
-	return this_room_type
-
+@export var this_room_event: RoomEvent
+@export var coordinates: Vector2
```

Comment:
i'm not exactly sure a room needs to know where it is on the map. That's more of an information for the map level I think